packages/client/src/admin/Apps/Apps.tsx

```diff
@@ -0,0 +1,265 @@
+    const descIndex = description
+      .toLowerCase()
+      .indexOf(searchText.toLowerCase());
+    return index >= 0 || descIndex >= 0;
+  };
+
+  React.useEffect(() => {
+    if (phAppTemplates.length > 0) {
+      setFilteredTemplates(phAppTemplates.filter(filter));
+    }
+  }, [phAppTemplates]);
+
+  React.useEffect(() => {
+    setFilteredTemplates(phAppTemplates.filter(filter));
+  }, [searchText]);
+
+  const onSearch = (text: string): void => {
+    setSearchText(text);
+  }
+
+  const PageHead = () => (
+    <div
+      style={{
+        background: '#fff',
+        borderBottom: '1px solid #eee',
+        padding: '16px 24px',
+      }}
+    >
+      <Breadcrumbs
+        pathList={[
+          {
+            key: 'apps',
+            matcher: /\/apps/,
+            title: 'Apps',
+          },
+        ]}
+      />
+    </div>
+  );
+
+  return (
+    <Layout>
+      <PageHead />
+      <div
+        style={{
+          margin: '16px 16px 0 16px',
+          padding: '32px',
+          backgroundColor: '#fff',
+        }}
+      >
+        <div
+          style={{
+            color: 'rgba(0, 0, 0, 0.85)',
+            paddingBottom: '16px',
+            fontWeight: 500,
+            fontSize: '16px',
+          }}
+        >
+          Import Custom App Template YAML from URL
+        </div>
+        <div
+          style={{
+            display: 'flex',
+            width: '100%',
+          }}
+        >
+          <Input
+            ref={importURL}
+            data-testid='import-template-url'
+            placeholder='Copy and Paste Application URL'
+          />
+          <Button
+            type='primary'
+            data-testid='import-template-button'
+            onClick={async () => {
+              let url = importURL.current.input.value;
+              if (url) {
+                url = url.replace(GITHUB_REGEX, GITHUB_REGEX_REPLACE_VALUE);
+                props.importPhAppTemplateFromURL({ variables: { url } });
+              }
+            }}
+            style={{ marginLeft: '16px' }}
+          >
+            Import
+          </Button>
+        </div>
+      </div>
+      <div
+        style={{
+          margin: '16px',
+          padding: '32px',
+          backgroundColor: '#fff',
+        }}
+      >
+        <div
+          style={{
+            color: 'rgba(0, 0, 0, 0.85)',
+            paddingBottom: '16px',
+            fontWeight: 500,
+            fontSize: '16px',
+          }}
+        >
+          Available Apps
+        </div>
+        <Row gutter={24} type='flex' style={{ marginBottom: '16px' }}>
+          <Col xs={24} md={12} xl={12} xxl={8}>
+            <Search
+              placeholder='Search application'
+              onChange={e => onSearch(e.currentTarget.value)}
+              onSearch={onSearch}
+            />
+          </Col>
+        </Row>
+        <Row gutter={24} type='flex'>
+          {filteredTemplates.map((appTemplate: PhAppTemplate): JSX.Element => {
```

Comment:
You don't need to put the type annotation in here, typescript will auto infer.

```suggestion
          {filteredTemplates.map((appTemplate: PhAppTemplate) => {
```